pkg/controller/elasticsearch/nodespec/volumes.go

```diff
@@ -114,7 +114,7 @@ func buildVolumes(
 		volumeMounts = append(volumeMounts, fileSettingsVolume.VolumeMount())
 	}
 
-	volumeMounts = esvolume.AppendDefaultDataVolumeMount(volumeMounts, volumes)
+	volumeMounts = esvolume.AppendDefaultDataVolumeMount(volumeMounts, append(volumes, nodeSpec.PodTemplate.Spec.Volumes...))
```

Comment:
Let's add a comment here to explain why we are suddenly including the user provided PodTemplate here.